src/main/java/guanguan/Gui.java

```diff
@@ -0,0 +1,92 @@
+package guanguan;
+
+/**
+ * Extension of Ui class that is responsible for the outputs on GUI.
+ */
+public class Gui extends Ui {
+    private final StringBuilder buffer;
+
+    /**
+     * Constructor for Gui.
+     */
+    public Gui() {
+        this.buffer = new StringBuilder();
+    }
+
+    /**
+     * Get accumulated output from the UI.
+     * @return Text output accumulated so far.
+     */
+    public String getTextOutput() {
```

Comment:
Maybe could add @override here? Or could remove ```getTextOutput()``` from ```Ui.java``` as it seems to do nothing